### DISCUSSION TITLE
How do I replace a specific part of a converted string?

### DISCUSSION THREAD

**Discussion by spifory** [markedjs/marked#3467] [Q&A] [answered]
What I want is for marked to turn a string, for example: 
```md
testing **image** ![test](...)
```

Into the following:
```html
testing **image** <img href="..." alt="test" />
```
Where it _only_ converts the image portion of the string into an HTML tag. (As I would like to replace what is in `href` alone)

As opposed to it currently doing:
```html
<p>testing <strong>image</strong> <img href="..." alt="test"/></p>
```

Is there a way to do this, or will I _have_ to let it convert everything in the string to HTML tags? Thank you

**Comment by UziTech** [comment, marked as answer]
Do you want to change the href in the middle of marked outputting html? In that case I would use use [walkTokens](https://marked.js.org/using_pro#walk-tokens) to look for link tokens and update the href accordingly.

If you actually only want link tokens to generate html, you can update all other [renderers](https://marked.js.org/using_pro#renderer) to just output the raw text instead of html.

```js
const renderer = {
  heading(token) {
    return token.raw;
  },
  ...
}
```

**Reply by spifory** [reply]
Yeah I want it to only generate HTML for link tokens so that I can grab the content of `token.href` and use it, while leaving every other token not HTML-ified. Second solution looks like it will do exactly what I need, thank you so much!
